client/windows/CWindowWithArtifacts.h

```diff
@@ -19,41 +19,44 @@
 class CWindowWithArtifacts : virtual public CWindowObject
 {
 public:
-	using CArtifactsOfHeroPtr = std::variant<
-		std::weak_ptr<CArtifactsOfHeroMarket>,
-		std::weak_ptr<CArtifactsOfHeroAltar>,
-		std::weak_ptr<CArtifactsOfHeroKingdom>,
-		std::weak_ptr<CArtifactsOfHeroMain>,
-		std::weak_ptr<CArtifactsOfHeroBackpack>,
-		std::weak_ptr<CArtifactsOfHeroQuickBackpack>>;
+	using ArtifactsOfHeroVar = std::variant<
+		std::shared_ptr<CArtifactsOfHeroMarket>,
+		std::shared_ptr<CArtifactsOfHeroAltar>,
+		std::shared_ptr<CArtifactsOfHeroKingdom>,
+		std::shared_ptr<CArtifactsOfHeroMain>,
+		std::shared_ptr<CArtifactsOfHeroBackpack>,
+		std::shared_ptr<CArtifactsOfHeroQuickBackpack>>;
```

Comment:
I know that this is not a new code, but why you opted to use variant here instead of using interface with different implementations with virtual methods?

Reply:
At a certain stage of working with the code, it was easier to distinguish which window we were working with at the moment. But for now, you may be right. Switching to virtual methods may make the code even simpler.

Reply:
I'll try to redo it